lib/vrf.c

```diff
@@ -57,6 +58,62 @@ static int vrf_backend;
 static int vrf_backend_configured;
 static char vrf_default_name[VRF_NAMSIZ] = VRF_DEFAULT_NAME_INTERNAL;
 
+/*Hash Table for table_id based lookup of vrf_id*/
+PREDECL_HASH(vrf_ids_by_table);
+
+struct vrf_ids_by_table_proxy {
+	struct vrf_ids_by_table_item vrf_ids_by_table_hash_item;
+	uint32_t table_id;
+	vrf_id_t vrf_id;
+};
+
+static struct vrf_ids_by_table_head vrf_ids_by_table_head;
+
+static int vrf_ids_by_table_cmp(const struct vrf_ids_by_table_proxy *item_a,
+				const struct vrf_ids_by_table_proxy *item_b)
+{
+	return (item_a->table_id == item_b->table_id);
+}
+
+static uint32_t vrf_ids_by_table_hash(const struct vrf_ids_by_table_proxy *item)
+{
+	return item->table_id;
```

Comment:
even though those table_id's should be unique, I think you still need to call `jhash_1word()` here to keep the table sparse.

Reply:
Yes, I agree. I looked for other uses of jhash_1word() in the codebase and followed suit. I generated and hard coded a random initval, which seems to be how it was done elsewhere.